src/editor/CodeHintList.js

```diff
@@ -65,7 +65,7 @@ define(function (require, exports, module) {
          *
          * @type {number}
          */
-        this.maxResults = ValidationUtils.isIntegerInRange(maxResults, 1, 1000) ? maxResults : 50;
+        this.maxResults = 500;
```

Comment:
This is already there you can change maxhints from preference. (i.e. "maxCodeHints": 100)

We are taking this value from preference only, Please check CodeHintManager.js
"maxCodeHints = PreferencesManager.get("maxCodeHints");"